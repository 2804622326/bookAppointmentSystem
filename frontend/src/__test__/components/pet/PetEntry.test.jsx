import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import PetEntry from '../../../components/pet/PetEntry.jsx';

const pet = { name: '', age: '', color: '', type: '', breed: '' };

test('calls removePet when remove button clicked', () => {
  const removePet = jest.fn();
  const { asFragment } = render(
    <PetEntry pet={pet} index={0} canRemove removePet={removePet} handleInputChange={() => {}} />
  );
  fireEvent.click(document.querySelector('button'));
  expect(removePet).toHaveBeenCalledWith(0);
  expect(asFragment()).toMatchSnapshot();
});
