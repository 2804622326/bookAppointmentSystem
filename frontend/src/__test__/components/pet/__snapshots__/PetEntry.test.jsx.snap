// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls removePet when remove button clicked 1`] = `
<DocumentFragment>
  <fieldset
    class="field-set mb-4"
  >
    <legend
      class="legend"
    >
      Pet #1 details
    </legend>
    <fieldset
      class="mb-4"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <input
            class="form-control"
            id="petName -0"
            name="name"
            placeholder="Enter pet name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="col-md-6"
        >
          <input
            class="form-control"
            id="petAge"
            name="age"
            placeholder="Enter pet age"
            required=""
            type="number"
            value=""
          />
        </div>
      </div>
    </fieldset>
    <div
      class="mb-4 col"
    >
      <div
        class="mb-2 col"
      >
        <select
          class="form-control"
          id="petColor"
          name="color"
          required=""
        >
          <option
            value=""
          >
            ...select pet color...
          </option>
          <option
            value="add-new-item"
          >
            Add New
          </option>
        </select>
      </div>
    </div>
    <fieldset
      class="field-set mb-4"
    >
      <legend
        class="legend"
      >
        Pet type and Breed
      </legend>
      <div
        class="mb-2 d-flex row"
      >
        <div
          class="col"
        >
          <div
            class="col"
          >
            <select
              class="form-control"
              id="petType"
              name="type"
              required=""
            >
              <option
                value=""
              >
                ...select type...
              </option>
              <option
                value="add-new-item"
              >
                Add New
              </option>
            </select>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="col"
          >
            <select
              class="form-control"
              id="petBreed"
              name="breed"
              required=""
            >
              <option
                value=""
              >
                ...select breed...
              </option>
              <option
                value="add-new-item"
              >
                Add New
              </option>
            </select>
          </div>
        </div>
      </div>
    </fieldset>
    <div
      class="d-flex justify-content-end mt-2"
    >
      <button
        class="btn btn-danger btn-sm"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
    </div>
  </fieldset>
</DocumentFragment>
`;
